src/framework/script/script-type.js

```diff
@@ -234,6 +260,57 @@ class ScriptType extends EventHandler {
         return this._enabled && !this._destroyed && this.entity.script.enabled && this.entity.enabled;
     }
 
+    /**
+     * Listen for an event on an EventHandler. The ScriptType will add and remove the event listeners on the
+     * EventHandler automatically when the ScriptType is enabled, disabled and destroyed.
+     *
+     * @param {EventHandler} eventHandler - EventHandler to listen for events on.
+     * @param {string} name - Name of the event to bind the callback to.
+     * @param {HandleEventCallback} callback - Function that is called when event is fired. Note
+     * the callback is limited to 8 arguments.
+     * @param {object} [scope] - Object to use as 'this' when the event is fired.
+     * @example
+     * this.listen(this.app.mouse, 'mousemove', this.onMouseMove, this);
+     * @see {@link ScriptType#unlisten} to remove listeners to an event on an EventHandler.
+     */
+    listen(eventHandler, name, callback, scope) {
+        this._listeners.push({
```

Comment:
```suggestion
        eventHandler.on("destroy", () => {
            this.unlisten(eventHandler, name, callback, scope);
        });
        this._listeners.push({
```

Reply:
That would have to assume that all objects have a destroy event which is not guaranteed

Reply:
> That would have to assume that all objects have a destroy event which is not guaranteed

Good point! Should we try to remove object references or accept that these objects will remain active in memory then?

As with everything, we have multiple possibilities:

1) Make sure to actually add destroy methods on obvious objects (good style anyway?)
2) Save some kind of GUID reference to objects, which isn't the actual object (would require GUID lookups etc.)
3) Simply accept we reference destroyed objects and trust the garbage collector to eventually free up the memory (as it is right now)
4...) ...any more ideas?

I guess option (3) would work fine in most cases, but this should be a conscious choice and not an oversight decision.

Reply:
Check out WeakMap and WeakSet.

Reply:
@Maksims Interesting suggestion, maybe I miss something, but we cannot iterate WeakMap and WeakSet by design, so how would we iterate over all event handlers to enable/disable them when needed?

Reply:
Thinking on it more, without at least periodical iteration through listened event handles, or an event on event handler - would be hard to implement self removals.